Dockerfile

```diff
@@ -0,0 +1,36 @@
+FROM ruby:3.0.6
+
+RUN apt-get update -y && apt-get install -y --no-install-recommends \
+  default-jre \
+  default-mysql-client \
+  imagemagick \
+  libprotobuf-dev \
+  protobuf-compiler \
+  nodejs yarn
+# Workaround for PhantomJS: https://github.com/DMOJ/online-judge/pull/1270
+ENV OPENSSL_CONF /etc/ssl/
+
+COPY Gemfile* /usr/src/app/
+WORKDIR /usr/src/app
+
+ENV BUNDLE_PATH /gems
+
+RUN bundle install
+
+ENV NODE_VERSION=16.13.0
+RUN apt install -y curl
```

Comment:
See if you can't add this `curl` to line 9 and delete 21 rather than having an additional RUN command here.

Also take a look at https://medium.com/@esotericmeans/optimizing-your-dockerfile-dc4b7b527756 to see if there are other optimizations you can make.